insights/parsers/httpd_conf.py

```diff
@@ -119,9 +121,9 @@ class HttpdConf(LegacyItemAccess, Parser):
 
     def __init__(self, *args, **kwargs):
         self.data = {}
-        self.full_data = {}
-        self.first_half = {}
-        self.second_half = {}
+        self.full_data = defaultdict(list)
+        self.first_half = defaultdict(list)
+        self.second_half = defaultdict(list)
```

Comment:
@skontar since these are type `defaultdict` they could inadvertently be modified if the rule checks for a key that does not exist.  One option is to convert these to `dict` once you have finished building them.